src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.EC.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class CngKey : IDisposable
+    {
+        /// <summary>
+        /// Does the key use elliptic curve cryptography
+        /// </summary>
+        /// <returns></returns>
+        internal bool IsEcc()
+        {
+            return (AlgorithmGroup == CngAlgorithmGroup.ECDiffieHellman ||
+                AlgorithmGroup == CngAlgorithmGroup.ECDsa);
+        }
+
+        /// <summary>
+        /// Does the key represent a named curve (Win10+)
+        /// </summary>
+        /// <returns></returns>
+        internal bool IsECNamedCurve()
+        {
+            return IsECNamedCurve(Algorithm.Algorithm);
+        }
+
+        internal static bool IsECNamedCurve(string algorithm)
+        {
+            return (algorithm == "ECDSA" || algorithm == "ECDH");
```

Comment:
Sometimes (here) we have the literal string repeated.  Others (GetEcSpecificCurveName) you dereference the CngAlgorithm static property value.  Why the inconsistency?